cmd/ftl/cmd_open.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"connectrpc.com/connect"
+	"github.com/alecthomas/kong"
+
+	ftlv1 "github.com/block/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/block/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
+	"github.com/block/ftl/common/reflection"
+	"github.com/block/ftl/common/schema"
+	"github.com/block/ftl/internal/exec"
+	"github.com/block/ftl/internal/log"
+	"github.com/block/ftl/internal/projectconfig"
+)
+
+type openCmd struct {
+	Ref    reflection.Ref `arg:"" help:"Language of the module to create." placeholder:"MODULE.ITEM" predictor:"decls"`
+	Editor string         `arg:"" help:"Editor to open the file with." enum:"auto,vscode,intellij" default:"auto"`
+
+	TerminalProgram  string `help:"Terminal program this command is running in which can influence 'auto' editor" env:"TERM_PROGRAM" hidden:""`
+	TerminalEmulator string `help:"Terminal emulator can influence 'auto' editor" env:"TERMINAL_EMULATOR" hidden:""`
```

Comment:
Wut?

Reply:
Yehhh just reading envars to determine if the user is inside VSCode or IntelliJ terminals. Each editor indicates it in different ways. Got a better idea?

Reply:
Ah I see. There's no good solution to this AFAIK, but I think this would be better if it was in a dedicated "editor" package that could detect the user's editor, or have it configured.